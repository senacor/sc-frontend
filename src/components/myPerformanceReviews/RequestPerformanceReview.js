import React, { useContext } from 'react';
import { connect } from 'react-redux';
import * as actions from '../../actions/index';
import Button from '@material-ui/core/Button';
import Icon from '@material-ui/core/Icon';
import Tooltip from '@material-ui/core/Tooltip';
import objectGet from 'object-get';
import { injectIntl } from 'react-intl';
import { Redirect } from 'react-router-dom';
import { UserinfoContext } from '../App';

export const RequestPerformanceReview = props => {
  let { newPrId, intl, addPr } = props;
  const { userinfo } = useContext(UserinfoContext.context).value;
  if (newPrId) {
    return <Redirect to={`/prDetail/${newPrId}`} />;
  }

  const hasSupervisor = objectGet(props, 'userinfo.hasSupervisor');
  const hasPrInProgress = objectGet(props, 'userinfo.hasPrInProgress');

  if (!hasSupervisor || hasPrInProgress) {
    let tooltipText = !hasSupervisor
      ? intl.formatMessage({
          id: 'requestperformancereview.nosupervisor'
        })
      : intl.formatMessage({
          id: 'requestperformancereview.alreadyopened'
        });

    return (
      <Tooltip title={tooltipText}>
        <span>
          <Button disabled>
            <Icon>add</Icon>
            {intl.formatMessage({
              id: 'requestperformancereview.requestpr'
            })}
          </Button>
        </span>
      </Tooltip>
    );
  }

  return (
    <React.Fragment>
      <Button
        id="addPrButton"
        color="primary"
        onClick={() => addPr(userinfo.userPrincipalName)}
        disabled={!hasSupervisor || hasPrInProgress}
      >
        <Icon>add</Icon>
        {intl.formatMessage({
          id: 'requestperformancereview.requestpr'
        })}
      </Button>
    </React.Fragment>
  );
};

export default injectIntl(
  connect(
    state => ({
      newPrId: state.newPrId
    }),
    { addPr: actions.addPr }
  )(RequestPerformanceReview)
);
