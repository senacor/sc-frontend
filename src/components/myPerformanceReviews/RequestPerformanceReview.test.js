import React from 'react';
import { shallow } from 'enzyme';
import { RequestPerformanceReview } from './RequestPerformanceReview';

describe('RequestPerformanceReview', () => {
  it('should match snapshot', () => {
    let cut = shallow(<RequestPerformanceReview />);

    expect(cut).toMatchSnapshot();
  });
});
