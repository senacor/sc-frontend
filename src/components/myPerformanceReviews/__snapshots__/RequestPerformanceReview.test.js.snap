// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestPerformanceReview should match snapshot 1`] = `
<WithStyles(Button)
  color="primary"
  id="addPrButton"
  onClick={[Function]}
>
  <WithStyles(Icon)>
    add
  </WithStyles(Icon)>
  PR beantragen
</WithStyles(Button)>
`;
